### PR TITLE
Execution state class

### PR DESCRIPTION
**Pull request by sugolov**
Static class reflecting program state designed to improve communication between CLI/Controller boundary,

### CODE REVIEW

**Comment by sugolov** [comment]
shit i forgot to pull the new changes from main before this lol

File: src/main/java/entities/ExecutionState.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+     * @variable allUserCourses represents all Courses initially instantiated by the user
+     * @variable remainingUserCourses represents all courses the user
+     */
+    private static boolean stateIsSetup;
+    private static GenerationMode generationMode;
+    private static List<Course> allUserCourses;
+    private static List<Course> remainingUserCourses;
+    private static Course currentCourse;
+
+    public ExecutionState(){}
```
Comment (kenneth-miura) [inline src/main/java/entities/ExecutionState.java:22]: imo just delete this constructor. It'll be auto-added on compile, and having a constructor (even an empty one) makes me think this class is supposed to be instantiated

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
Looks good! One change req. Also, u should add this to design doc as example of clean architecture

File: src/main/java/entities/ExecutionState.java (hunk 1 above)
Comment (sugolov) [reply src/main/java/entities/ExecutionState.java:22]: Yeah true

**Comment by sugolov** [comment]
Will do

**Pull request merged**